src/components/banner/index.tsx

```diff
@@ -82,93 +84,127 @@ export const Banner = (props: EmbeddedMessageData) => {
-                  clickedUrl
-                );
-                embeddedManager.trackEmbeddedClick(
-                  message,
-                  button?.id,
-                  clickedUrl
-                );
-              }}
-            >
-              {button.title ? button.title : `Button ${index + 1}`}
-            </button>
-          ))}
+  const handleButtonClick = (button: any) => {
+    const clickedUrl =
+      button?.action?.data?.trim() || button?.action?.type || '';
+    embeddedManager.handleEmbeddedClick(message, button?.id, clickedUrl);
+    embeddedManager.trackEmbeddedClick(message, button?.id, clickedUrl);
+  };
+
+  function addButtonClickEvent(button: HTMLElement, index: number) {
+    button.addEventListener('click', (event) => {
+      // Prevent the click event from bubbling up to the div
+      event.stopPropagation();
+      if (!message?.elements?.buttons) {
+        return '';
+      }
+      handleButtonClick(message?.elements?.buttons[index]);
+    });
+  }
+
+  setTimeout(() => {
+    const bannerDiv = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner`
+    )[0];
+    const primaryButtonClick = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner-primaryButton`
+    )[0];
+    const secondaryButtonClick = document.getElementsByName(
+      `${message?.metadata?.messageId}-banner-secondaryButton`
+    )[0];
+    if (bannerDiv) {
+      bannerDiv.addEventListener('click', handleBannerClick);
+    }
+    if (primaryButtonClick) {
+      addButtonClickEvent(primaryButtonClick, 0);
+    }
+    if (secondaryButtonClick) {
+      addButtonClickEvent(secondaryButtonClick, 1);
+    }
+  }, 0);
+
+  return `
+    <style>${mediaStyle}</style>
+    <div 
+      class="banner" 
+      id="${parentId}"
+      name="${message?.metadata?.messageId}-banner"
+      style="${defaultBannerStyles}; ${parentStyle || ''}" 
+    >
+      <div class="banner" 
+      style="display: flex; flex-direction: row;">
+        <div class="banner" 
+        style="${defaultTextParentStyles}">
+          <text class="titleText banner"  id="${titleId}" style="${defaultTitleStyles}; ${
+    titleStyle || ''
+  }">
+            ${message?.elements?.title || 'Title Here'}
+          </text>
+          <text class="titleText banner" id="${textId}" style="${defaultTextStyles}; ${
+    textStyle || ''
+  }">
+            ${message?.elements?.body}
+          </text>
         </div>
+        ${
+          message?.elements?.mediaUrl
+            ? `<img class="banner" id="${imageId}"
+            style="${defaultImageStyles}; ${imgStyle || ''}" src="${
+                message?.elements?.mediaUrl
+              }" />`
+            : ''
+        }
+      </div>
+      <div class="banner" 
+       style="${bannerButtons}">
+        ${message?.elements?.buttons
+          ?.map((button: any, index: number) => {
+            const buttonStyle =
+              index === 0 ? primaryBtnStyle : secondaryBtnStyle;
```

Comment:
we have a lot of logic for the styles based on primary/secondary button. 

we should create an object that handles this outside of the component string to clean it up a bit. It would look sort of like:

```
const getObj = () => ({
   buttonStyle: index === 0 ? primaryBtnStyle : secondaryBtnStyle,
   disablePrimaryBtn: index === 0 ? 'disabled' : 'enabled',
   ...
})
```

Reply:
Done